bowtie/_cli.py

```diff
@@ -1271,6 +1277,179 @@ def validate(
+    default=1,
+    show_default=True,
+    help="Number of skipped values per run used to warmup the benchmark.",
+)
+@click.option(
+    "--loops",
+    "-l",
+    "loops",
+    type=int,
+    default=1,
+    show_default=True,
+    help="Number of loops per value.",
+)
+@click.option(
+    "--quiet",
+    "-q",
+    "quiet",
+    is_flag=True,
+    default=False,
+    show_default=True,
+    help="Enable quiet mode (Only output the final result).",
+)
+@click.option(
+    "--keywords",
+    "-k",
+    "keywords",
+    is_flag=True,
+    default=False,
+    show_default=True,
+    help=(
+        "Run keyword specific benchmarks to learn about how "
+        "various implementations implement the keyword."
+    ),
+)
+@click.option(
+    "-b",
+    "--benchmark-file",
+    "benchmark_files",
+    multiple=True,
+    help=(
+        "Allows running benchmark from a file. "
+        "Specify the path of the benchmark file"
+    ),
+)
+@click.option(
+    "--test-suite",
+    "-t",
+    "test_suite",
+    type=_suite.ClickParam(),
+    default=None,
+    help="Run Benchmarks over the official JSON Schema Test Suite.",
+)
+@click.argument("benchmark", type=JSON(), required=False)
+def perf(
+    connectables: Iterable[_connectables.Connectable],
+    dialect: Dialect,
+    format: _F,
+    keywords: bool,
+    benchmark: dict[str, Any] | None,
+    quiet: bool,
+    test_suite: tuple[Iterable[TestCase], Dialect, dict[str, Any]],
+    benchmark_files: Iterable[str],
+    **kwargs: Any,
+):
+    """
+    Perform performance measurements across supported implementations.
+    """
+    try:
+        if benchmark_files:
+            benchmarker = _benchmarks.Benchmarker.for_benchmark_files(
+                benchmark_files,
+                **kwargs,
+            )
+        elif keywords:
+            benchmarker = _benchmarks.Benchmarker.for_keywords(
+                dialect,
+                **kwargs,
+            )
+        elif test_suite:
+            cases, enforced_dialect, _ = test_suite
+            benchmarker = _benchmarks.Benchmarker.from_test_cases(
+                cases,
+                **kwargs,
+            )
+            dialect = enforced_dialect
+        elif benchmark is None:
+            benchmarker = _benchmarks.Benchmarker.from_default_benchmarks(
+                **kwargs,
+            )
+        else:
+            benchmarker = _benchmarks.Benchmarker.from_input(
+                benchmark=benchmark,
+                **kwargs,
+            )
+
+        asyncio.run(
+            benchmarker.start(
+                connectables=connectables,
+                dialect=dialect,
+                quiet=quiet,
+                format=format,
+            ),
+        )
+    except (BenchmarkError, BenchmarkLoadError) as err:
+        STDERR.print(err)
+        return EX.DATAERR
+
+    return 0
+
+
+@subcommand
+@dialect_option()
+@click.option(
+    "-t",
+    "--benchmark-type",
+    type=click.Choice(["default", "keyword"]),
```

Comment:
Doesn't have to be immediately (i.e. we can leave it and "fix"/"change") but I think `tags` or `markers` makes a bit more sense (i.e. that we have benchmarks have some notion of tags [a la website] or markers [a la pytest test markers]) and then you can filter by a tag/marker, and then right now we just have 1 of those, for keyword benchmarks.

Reply:
I'll open an issue for this, and if time permits then would add this functionality.